internal/schema/schema.go

```diff
@@ -0,0 +1,340 @@
+package schema
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/qri-io/jsonschema"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	enginev1 "github.com/cerbos/cerbos/api/genpb/cerbos/engine/v1"
+	schemav1 "github.com/cerbos/cerbos/api/genpb/cerbos/schema/v1"
+	"github.com/cerbos/cerbos/internal/config"
+	"github.com/cerbos/cerbos/internal/observability/logging"
+	"github.com/cerbos/cerbos/internal/storage"
+)
+
+type Manager struct {
+	conf                *Conf
+	log                 *zap.SugaredLogger
+	store               storage.Store
+	mu                  sync.RWMutex
+	principalProperties map[string]string
+	resourceProperties  map[string]map[string]string
+	schema              *schemav1.Schema
+	principalSchema     *jsonschema.Schema
+	resourceSchemas     map[string]*jsonschema.Schema
+}
+
+func New(store storage.Store) (*Manager, error) {
+	conf := &Conf{}
+	if err := config.GetSection(conf); err != nil {
+		return nil, fmt.Errorf("failed to get config section %q: %w", confKey, err)
+	}
```

Comment:
nit: Usually it's easier for reader to use address operator when passing pointers to functions such as `config.GetSection(&conf)`.